clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1822,6 +1838,90 @@ static std::string getCPPTypeString(QualType Ty) {
   return eraseAnonNamespace(Ty.getAsString(P));
 }
 
+static void printArguments(ASTContext &Ctx, raw_ostream &ArgOS,
+                           ArrayRef<TemplateArgument> Args,
+                           const PrintingPolicy &P, bool ParameterPack) {
+
+  if (!ParameterPack)
+    ArgOS << "<";
+
+  bool FirstArg = true;
+  const char *Comma = ", ";
+
+  for (unsigned I = 0; I < Args.size(); I++) {
+    const TemplateArgument &Arg = Args[I];
+    if (Arg.getKind() == TemplateArgument::ArgKind::Pack) {
```

Comment:
This seems a lot like a few of the other places we emit arguments based on their kind.  Is there a way to dispatch the body of much of this instead?

Reply:
Maaaaybe. But I think the function body is not similar enough to make this efficient. Both functions are looping over arguments to emit 'something' but what is being emitted is different.